lib/lightning_web/live/components/form.ex

```diff
@@ -305,11 +305,12 @@ defmodule LightningWeb.Components.Form do
         error_tag_classes: error_tag_classes,
         label_classes: label_classes
       )
+      |> assign_new(:disabled, fn -> false end)
 
     ~H"""
     <div class="flex items-start">
       <div class="flex items-center h-5">
-        <%= checkbox(@form, @id, class: @checkbox_classes) %>
+        <%= checkbox(@form, @id, disabled: @disabled, class: @checkbox_classes) %>
       </div>
       <div class="ml-3 text-sm">
```

Comment:
Instead of making a new assigns to fill a keyword list (which is what `checkbox/3` takes at the end).

Liveview has some conventions around this:

```
 attr :opts, :global   # <== This filters out all the standard HTML attrs (like disabled)

 def label_field(assigns) do
 ....

    assigns =
      assigns
      |> update(:opts, fn opts ->
        assigns_to_attributes(opts)
        |> Keyword.put_new(:class, label_classes)
      end)

     ~H"""
    <%= label(@form, @id, @title, @opts) %>
     """
   end
```

These are some changes I have for `label_field` in my current branch, I wanted to be able to override the class, but it will work with `disabled` because it's a standard HTML attribute.

And `assigns_to_attributes/2` turns assigns into a keyword list allowing us to easily drop the resulting valid in `label/4` or in your case `checkbox/3`.

Check out: https://hexdocs.pm/phoenix_live_view/Phoenix.Component.html#assigns_to_attributes/2
and https://hexdocs.pm/phoenix_live_view/Phoenix.Component.html#attr/3

The `attr/3` function was introduced not so long ago so I've only just started adding it to our components as I go.